pkg/sql/planner.go

```diff
@@ -710,6 +712,11 @@ func (p *planner) IsActive(ctx context.Context, key clusterversion.Key) bool {
 	return p.execCfg.Settings.Version.IsActive(ctx, key)
 }
 
+// GetTenantCapabilitiesCache is part of the eval.Planner interface.
+func (p *planner) GetTenantCapabilitiesCache() map[roachpb.TenantID]tenantcapabilitiespb.TenantCapabilities {
+	return p.ExecCfg().TenantCapabilitiesReader.GetCapabilitiesMap()
+}
```

Comment:
I believe this will crash currently in a non-system-tenant sql servers because we aren't initialising the TenantCapabilitiesReader in that case.  

I think it is probably correct that we don't provide an easy means for non-system-tenants to read from this cache since it exposes information from all tenants, but I imagine we don't want it to crash.  Perhaps we return an empty map if it isn't initialized?

Reply:
Also added a check for having the admin role like `SHOW TENANT`